src/main/java/uk/gov/hmcts/ccd/domain/model/definition/SearchResultField.java

```diff
@@ -108,4 +119,32 @@ public String getDisplayContextParameter() {
     public void setDisplayContextParameter(String displayContextParameter) {
         this.displayContextParameter = displayContextParameter;
     }
+
+    public UseCase getUseCase() {
+        return useCase;
+    }
+
+    public void setUseCase(UseCase useCase) {
+        this.useCase = useCase;
+    }
+
+    public Object getObjectByPath(JsonNode value) {
+        List<String> pathElements = getCaseFieldPathElements();
+
+        return reduce(value, pathElements, getCaseFieldPath());
+    }
+
+    private Object reduce(JsonNode caseFields, List<String> pathElements, String path) {
+        String firstPathElement = pathElements.get(0);
+
+        JsonNode caseField = Optional.ofNullable(caseFields.get(firstPathElement))
+            .orElseThrow(() -> new BadRequestException(format(NESTED_ELEMENT_NOT_FOUND_FOR_PATH, path)));
+
+        if (pathElements.size() == 1) {
+            return caseField;
+        } else {
+            List<String> tail = pathElements.subList(1, pathElements.size());
+            return reduce(caseField, tail, path);
+        }
+    }
```

Comment:
temporary: not sure this is the best place for this code. Maybe better moving this to a new method in `CaseFieldPathUtils`?